packages/dd-trace/src/profiling/profilers/events.js

```diff
@@ -171,10 +171,11 @@ class EventsProfiler {
   stop () {
     if (this._observer) {
       this._observer.disconnect()
+      this._observer = undefined
     }
   }
 
-  profile (startDate, endDate) {
+  profile ({ startDate, endDate, restart = true }) {
```

Comment:
I wanted to ask what's the value in passing an object here, and what's the value in having a default for restart – if the only call site is in profiler.js and it always passes all three values then just keeping 3-args would suffice. Except if you want to avoid specifying all args in wall and space profilers, in which case this might make sense. I'm still unclear on what does the default buy us.

Reply:
Well, my initial thought was that since wall and space profilers did not use startDate/endDate, it would be nice to not specify these parameters for them and at the same time make the API flexible.
But since this can be considered as an internal API than we can change at will, I guess we can pass 3 arguments as well.

Reply:
But having 3 args, implies updating tests with things like `profiler.profile(undefined, undefined, true)`

Reply:
I could swap the args to avoid that:
`profile (restart, startDate, endDate)`